Services/Music_Portal.Services.Services/ArtistService.cs

```diff
@@ -0,0 +1,37 @@
+﻿using System.Collections.Generic;
+using System.Linq;
+using System.Threading.Tasks;
+using AutoMapper;
+using Music_Portal.Domain.Core;
+using Music_Portal.Domain.Interfaces;
+using Music_Portal.Services.Interfaces;
+
+namespace Music_Portal.Services.Services
+{
+    public class ArtistService : IArtistService
+    {
+        private readonly ILastFmService _lastFmService;
+        private readonly IMapper _mapper;
+        private readonly IArtistRepository _repository;
+
+        public ArtistService(ILastFmService lastFmService, IMapper mapper, IArtistRepository repository)
+        {
+            _lastFmService = lastFmService;
+            _mapper = mapper;
+            _repository = repository;
+        }
+
+        public async Task<IEnumerable<Artist>> GetTopArtists()
+        {
+            var artists = _repository.GetArtistsEnumerable();
+            if (artists == null)
+            {
+                var topArtistsLastFm = await _lastFmService.GetTopArtists();
+                var topArtists = _mapper.Map<IEnumerable<Artist>>(topArtistsLastFm);
+                _repository.CreateRange(topArtists);
```

Comment:
Hmm your first call to add any artist will return break you call, in the end of the `if` you need to add `return` to fix that problem